src/utils/common.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Function to stripe array brackets if string defining the type has it.
+ *
+ * @param typeString - String defining type from which array brackets are required to be removed.
+ * @returns Parameter string with array brackets [] removed.
+ */
+export const stripArrayTypeIfPresent = (typeString: string) => {
+  if (typeString?.match(/\S\s*\[\]$/u) !== null) {
+    return typeString.replace(/\s*\[\]$/gu, '').trim();
```

Comment:
To address the non-compliant `Type []` case:
```suggestion
  if (typeString?.match(/\S\[\]$/u) !== null) {
    return typeString.replace(/\[\]$/gu, '').trim();
```